modules/standard/IO.chpl

```diff
@@ -5551,7 +5602,11 @@ proc _setIfPrimitive(ref lhs:?t, rhs:?t2, argi:int):syserr where t==bool&&_isIoP
 private inline
 proc _setIfPrimitive(ref lhs:?t, rhs:?t2, argi:int):syserr where t!=bool&&_isIoPrimitiveType(t)
 {
-  lhs = rhs:t;
+  try {
+    lhs = rhs:t;
+  } catch {
+    return EFORMAT;
```

Comment:
@mppf: I'm also not confident with this error code, any suggestions?

Reply:
I think EFORMAT is reasonable here, but so is ERANGE (as the comment above points out)